crates/networking/rpc/eth/transaction.rs

```diff
@@ -436,15 +436,27 @@ impl RpcHandler for EstimateGasRequest {
             // Block not found
             _ => return Ok(Value::Null),
         };
+
+        let transaction = if self.transaction.nonce.is_none() {
```

Comment:
Nit: It's more idiomatic to use a match, but there's nothing wrong with this if.

Reply:
Which makes sense, I'll stick to the more idiomatic way!

Reply:
Done!